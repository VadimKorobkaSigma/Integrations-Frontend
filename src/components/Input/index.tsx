import * as React from 'react';
import cn from 'classnames';
import SVG, { Props } from 'react-inlinesvg';

import styles from './styles.module.scss';

export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
    icon?: string;
    containerClassname?: string;
    containerAttributes?: React.HTMLAttributes<HTMLDivElement>;
    iconAttributes?: Props;
}

const Input: React.FC<InputProps> = ({ icon, containerAttributes, containerClassname, ...args }) => {
    return (
        <div
            {...{ ...containerAttributes }}
            className={cn(styles.container, containerClassname, containerAttributes?.className)}
        >
            <input {...{ ...args }} className={cn(icon && styles.withIcon, args.className)} />
            {icon && <SVG {...{ ...args.iconAttributes, src: icon }} />}
        </div>
    );
};

export default React.memo(Input);
