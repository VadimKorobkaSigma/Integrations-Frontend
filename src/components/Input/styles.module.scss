.container {
  width: 300px;
  height: 40px;
  position: relative;

  svg {
    height: 24px;
    width: 24px;
    position: absolute;
    left: 0;
    top: 50%;
    transform: translateY(-50%);
    margin: 0 3px;
  }

  input {
    padding: 7px;
    width: 100%;
    height: 100%;
    border: 1px solid $grey-light;
    border-radius: 4px;

    &.withIcon {
      padding-left: 30px;
    }

    &:active,
    &:hover,
    &:focus {
      border-color: $blue;

      & + svg path {
        fill: $blue;
      }
    }
  }
}
