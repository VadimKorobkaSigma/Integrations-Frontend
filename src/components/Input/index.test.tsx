import * as React from 'react';
import Input from './index';
import { shallow } from 'enzyme';

test('Input Component without error', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Input onChange={onChange} />);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.find('input').simulate('change', 'New value');
    expect(onChange).toHaveBeenCalledWith('New value');
    wrapper.setProps({ containerAttributes: { className: 'testClass' } });
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.setProps({ icon: 'testIcon' });
    expect(wrapper.html()).toMatchSnapshot();
});
