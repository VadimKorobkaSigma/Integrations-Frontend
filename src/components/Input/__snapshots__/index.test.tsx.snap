// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component without error 1`] = `"<div class=\\"\\"><input class=\\"\\"/></div>"`;

exports[`Input Component without error 2`] = `"<div class=\\"testClass\\"><input class=\\"\\"/></div>"`;

exports[`Input Component without error 3`] = `"<div class=\\"testClass\\"><input class=\\"\\"/></div>"`;
